Zend/zend_vm_execute.h

```diff
@@ -3905,6 +3906,75 @@ static ZEND_OPCODE_HANDLER_RET ZEND_FASTCALL ZEND_DEFINED_SPEC_CONST_HANDLER(ZEN
 	ZEND_VM_NEXT_OPCODE();
 }
 
+static ZEND_OPCODE_HANDLER_RET ZEND_FASTCALL ZEND_FUNC_EXISTS_SPEC_CONST_HANDLER(ZEND_OPCODE_HANDLER_ARGS)
+{
+	USE_OPLINE
+	zend_function *fbc;
+
+	SAVE_OPLINE();
+	if (IS_CONST == IS_CONST) {
+		switch ((uintptr_t) CACHED_PTR(Z_CACHE_SLOT_P(EX_CONSTANT(opline->op1)))) {
+			case 0x1:
+				ZVAL_FALSE(EX_VAR(opline->result.var));
+				break;
+
+			case 0x0:
+				if ((fbc = zend_hash_find_ptr(EG(function_table), Z_STR_P(EX_CONSTANT(opline->op1)))) == NULL) {
+					ZVAL_FALSE(EX_VAR(opline->result.var));
+					CACHE_PTR(Z_CACHE_SLOT_P(EX_CONSTANT(opline->op1)), (void *) 0x1);
+					break;
+				}
+				CACHE_PTR(Z_CACHE_SLOT_P(EX_CONSTANT(opline->op1)), fbc);
+				/* intentionally missing break */
+			default:
+				ZVAL_TRUE(EX_VAR(opline->result.var));
+				break;
+		}
+	} else do {
+
+		zval *value = EX_CONSTANT(opline->op1);
+		zend_string *name, *lcname;
+
+try_func_ex:
+		if (EXPECTED(Z_TYPE_P(value) == IS_STRING)) {
+			name = Z_STR_P(value);
+		} else {
+			if (IS_CONST & (IS_CV|IS_VAR) && Z_TYPE_P(value) == IS_REFERENCE) {
+				value = Z_REFVAL_P(value);
+				goto try_func_ex;
+			}
+			if (Z_TYPE_P(value) < IS_STRING) {
+				if (Z_COPYABLE_P(value) && Z_REFCOUNT_P(value) > 1) {
```

Comment:
hmm… There's something a bit incorrect with reference handling. the refcount should be on the original value.